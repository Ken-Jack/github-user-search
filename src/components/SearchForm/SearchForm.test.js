import React from 'react';
import {shallow} from 'enzyme';
import felaSnapshot from 'test-util/fela-snapshot';

import DefaultSearchForm, {
  SearchForm,
} from './SearchForm';

describe('Component: SearchForm', () => {

  it('should render as expected', () => {
    const component = felaSnapshot(<DefaultSearchForm onSubmit={jest.fn()} />);
    expect(component).toMatchSnapshot();
  });

  describe('when an initialInputValue prop is absent', () => {
    it('should set inputValue to an empty string', () => {
      const wrapper = shallow(<SearchForm onSubmit={jest.fn()} />);
      expect(wrapper.state('inputValue')).toEqual('');
    });
  });

  describe('when a different initialInputValue is set', () => {
    it('should update the state', () => {
      const wrapper = shallow(
        <SearchForm
          initialInputValue={'hello'}
          onSubmit={jest.fn()}
        />
      );
      expect(wrapper.state('inputValue')).toEqual('hello');
      wrapper.setProps({
        initialInputValue: 'foo',
      });
      expect(wrapper.state('inputValue')).toEqual('foo');
    });
  });

  describe('when the form is submitted', () => {
    describe('and the state has an input value', () => {
      it('should pass the value to the onSubmit function', () => {
        const spy = jest.fn();
        const wrapper = shallow(
          <SearchForm
            onSubmit={spy}
          />
        );
        wrapper.setState({inputValue: 'test'});
        wrapper.simulate('submit', {preventDefault: f => f});
        expect(spy.mock.calls).toMatchSnapshot();
      });
    });

    describe('and the state has no inputValue', () => {
      it('should not call the onSubmit function', () => {
        const spy = jest.fn();
        const wrapper = shallow(
          <SearchForm
            onSubmit={spy}
          />
        );
        wrapper.simulate('submit', {preventDefault: f => f});
        expect(spy).not.toHaveBeenCalled();
      });
    });
  });

});
