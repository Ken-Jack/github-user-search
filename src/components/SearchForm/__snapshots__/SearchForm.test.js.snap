// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SearchForm should render as expected 1`] = `
Object {
  "component": <form
    className={[Function]}
    onSubmit={[Function]}
  >
    <label
      className="u-hiddenVisually"
      htmlFor="searchInput"
    >
      Search for a user
    </label>
    <input
      className={[Function]}
      id="searchInput"
      onChange={[Function]}
      placeholder="Search for a user, e.g. simonsmith"
      type="text"
      value=""
    />
    <button
      className={[Function]}
      type="submit"
    >
      <svg />
      <span
        className="u-hiddenVisually"
      >
        Submit
      </span>
    </button>
  </form>,
  "styles": "
    .a{position:relative}
    .b{padding:8px}
    .c{padding-right:40px}
    .d{width:100%}
    .e{border:1px solid #bbb}
    .f{box-shadow:inset 0 2px 2px rgba(0, 0, 0, .1);}
    .g{padding:3px 14px}
    .h{position:absolute}
    .i{top:4px}
    .j{right:0}",
}
`;

exports[`Component: SearchForm when the form is submitted and the state has an input value should pass the value to the onSubmit function 1`] = `
Array [
  Array [
    "test",
  ],
]
`;
